pkg/cache/clusterqueue_snapshot.go

```diff
@@ -28,13 +27,11 @@ type ClusterQueueSnapshot struct {
 	// In case its empty, it means an AdmissionCheck should apply to all ResourceFlavor
 	AdmissionChecks map[string]sets.Set[kueue.ResourceFlavorReference]
 	Status          metrics.ClusterQueueStatus
-	Quotas          map[resources.FlavorResource]*ResourceQuota
-	// GuaranteedQuota records how much resource quota the ClusterQueue reserved
-	// when feature LendingLimit is enabled and flavor's lendingLimit is not nil.
-	GuaranteedQuota resources.FlavorResourceQuantities
 	// AllocatableResourceGeneration will be increased when some admitted workloads are
 	// deleted, or the resource groups are changed.
 	AllocatableResourceGeneration int64
+
+	ResourceNode ResourceNode
 }
 
```

Comment:
```suggestion
var _ dominantResourceShareNode = (*ClusterQueueSnapshot)(nil)
```

@gabesaba 
Regarding the lint error, how about this obvious interface implementation?

Reply:
Additionally, it would be great to check if bumping golangci-lint version resolve this issue: https://github.com/kubernetes-sigs/kueue/pull/2857

Reply:
This gives the same error: have Parent() *CohortSnapshot, want Parent() lendableNode

I was able to get rid of the generics, and define methods on `clusterQueue` and `ClusterQueueSnapshot` to return the `lendableNode` type

Reply:
> Additionally, it would be great to check if bumping golangci-lint version resolve this issue: #2857

I resolved the issue, but I will leave as it is, without generics, as I think this is easier to read